addons/hr_timesheet/models/hr_timesheet.py

```diff
@@ -64,6 +65,8 @@ def _timesheet_preprocess(self, vals):
         if vals.get('project_id') and not vals.get('account_id'):
             project = self.env['project.project'].browse(vals.get('project_id'))
             vals['account_id'] = project.analytic_account_id.id
+            if not project.analytic_account_id.active:
```

Comment:
[Commit4] Small question while we are at it: is there any logic and/or propagation implemented about archiving stuff ? Like archiving a project, unarchiving a project, archiving analytic account, ... ?

EDIT: I see there are stuff displayed on project and task views, so I suppose it is ok and mighty bst and you took it into consideration :) .